src/libsyntax/parse/mod.rs

```diff
@@ -148,7 +148,7 @@ pub fn parse_stmt_from_source_str(name: String,
         name,
         source
     );
-    maybe_aborted(p.parse_stmt(), p)
+    maybe_aborted(panictry!(p.parse_stmt_nopanic()), p)
```

Comment:
The idea is that the error handling will eventually be pushed out another step, and `panictry!` will become `try!`, at least for most of the users.  Also, I'm planning to get rid of the panicking versions in a followup; then we can rename the `_nopanic` functions so the result looks less strange.